testcases/rpmTests/requires_test.py

```diff
@@ -0,0 +1,244 @@
+import utils.pkg_name_split as ns
+import utils.core.base_xtest as bt
+import config.runtime_config as rc
+import outputControl.logging_access as la
+import utils.process_utils as pu
+import utils.test_utils as tu
+import utils.core.configuration_specific as cs
+
+import sys
+
+
+class Default(cs.JdkConfiguration):
+    def __init__(self, this):
+        super().__init__()
+        self.rpms = rc.RuntimeConfig().getRpmList()
+        self.this = this
+
+    def _get_artificial_requires(self, filename):
+        output, error, res = pu.executeShell(f"rpm -qp rpms/{filename} --requires")
+        lines = output.split("\n")
+        return lines
+
+    def check_artificial_requires(self, this):
+        files = self.rpms.files
+        documentation = ""
+        files = [x.replace("rpms/", "") for x in files]
+        if not self.documenting:
+            files = [x for x in files if tu.validate_arch_for_rpms(this.current_arch) == ns.get_arch(x)]
+        for filename in files:
+            filename = filename.split("/")[-1]
+            expected_requires = self._get_expected_articial_requires(filename)
+            documentation += make_rpm_readable(filename) + " should have these artificial requires: " + ", ".join(expected_requires) + "\n"
+            given_requires = self._get_artificial_requires(filename)
+            missing_requires = []
+            for require in expected_requires:
+                if not tu.passed_or_failed(self, self.is_in_requires(require, given_requires), make_rpm_readable(filename) + " is missing reuqire: " + require):
```

Comment:
it is unneccessary to run the test while documenting, which is exactly what is happening here